external/source/include/windows/definitions.h

```diff
@@ -7,6 +7,13 @@
 typedef long NTSTATUS;
 #endif
 
+typedef struct _BIG_POOL_INFO {
+	PVOID Address;
+	DWORD64 PoolSize;

```

Comment:
Is this the correct type for 32bit systems as well? I can't seem to find a definitive reference online.

Reply:
The official reference I could find was at https://www.alex-ionescu.com/?p=231 which notes the following structure:

```
lkd> dt nt!_POOL_TRACKER_BIG_PAGES
    +0x000 Va : Ptr32 Void
    +0x004 Key : Uint4B
    +0x008 PoolType : Uint4B
    +0x00c NumberOfBytes : Uint4B
```

On x64 systems here is what it looks like:

```
lkd>  dt nt!_POOL_TRACKER_BIG_PAGES
   +0x000 Va               : Uint8B
   +0x008 Key              : Uint4B
   +0x00c Pattern          : Pos 0, 8 Bits
   +0x00c PoolType         : Pos 8, 12 Bits
   +0x00c SlushSize        : Pos 20, 12 Bits
   +0x010 NumberOfBytes    : Uint8B
```

Another frequently referenced blog was https://blahcat.github.io/2019/03/17/small-dumps-in-the-big-pool/ which shows the following structure:

```
0x00 NumberOfEntries

Entry0
{
0x08 ULONG_PTR Entry0.Address
0x10 DWORD Entry0.PoolSize
0x18 DWORD Entry0.PoolTag
}

Entry1
{
0x20 Entry1.Address
[...]
```

Reply:
I'm going to take this out and put this in my own file since quite honestly I'm not sure what the true definition is here or if this makes sense and until I do it doesn't make sense altering this at such a global level.